### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by adiliqbl**

### CODE REVIEW

File: app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImageDetailsFragment.kt [hunk 1]
```diff
@@ -14,17 +21,38 @@ class ImageDetailsFragment : BaseFragment<ImageDetailsViewModel>() {
 
     companion object {
         private const val KEY_IMAGE = "image"
+        private const val KEY_TRANSITION_NAME = "transitionName"
 
         @JvmStatic
```
Comment (osamaanwar94) [inline app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImageDetailsFragment.kt]: @adiliqbl JvmStatic should only be used if you are sure that your code will be accessed from a Java file

File: app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImageDetailsFragment.kt [hunk 2]
```diff
@@ -14,17 +21,38 @@ class ImageDetailsFragment : BaseFragment<ImageDetailsViewModel>() {
...
-        fun newInstance(image: Image): ImageDetailsFragment {
+        fun newInstance(image: Image, transitionName: String): ImageDetailsFragment {
```
Comment (osamaanwar94) [inline app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImageDetailsFragment.kt:32]: @adiliqbl Making an extension function in the BaseActivity/BaseFragment which would create and apply the bundle to the new activity/fragment instance, would reduce the code

File: app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImagesFragment.kt [hunk 3]
```diff
@@ -60,13 +60,13 @@ class ImagesFragment : BaseFragment<ImagesViewModel>() {
...
         imagesAdapter = ImagesPagedAdapter(glide, object : ImagesPagedAdapter.ImagesListener {
-            override fun onClick(sharedView: ImageView, image: Image) {
-                val fragmentTransaction = fragmentManager!!.beginTransaction()
-                fragmentTransaction.setCustomAnimations(android.R.anim.slide_in_left, android.R.anim.slide_out_right)
-                fragmentTransaction.replace((view.parent as ViewGroup).id, ImageDetailsFragment.newInstance(image), tag)
-                fragmentTransaction.addToBackStack(tag)
-                fragmentTransaction.addSharedElement(sharedView, getString(R.string.transition_image))
-                fragmentTransaction.commit()
+            override fun onClick(holder: ImagesPagedAdapter.ViewHolder, image: Image, transitionName: String) {
+                fragmentManager!!
```
Comment (osamaanwar94) [inline app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImagesFragment.kt:64]: @adiliqbl Whenever possible make extensions functions as they can reduce the copy/paste code from different classes

File: app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImagesPagedAdapter.kt [hunk 4]
```diff
@@ -18,7 +19,7 @@ import com.bumptech.glide.request.RequestOptions
 class ImagesPagedAdapter(private val glide: RequestManager, private val listener: ImagesListener) : PagedListAdapter<Image, ImagesPagedAdapter.ViewHolder>(Image.DIFF_CALLBACK) {
 
     interface ImagesListener {
-        fun onClick(sharedView: ImageView, image: Image)
+        fun onClick(holder: ImagesPagedAdapter.ViewHolder, image: Image, transitionName: String)
```
Comment (osamaanwar94) [inline app/src/main/java/com/adiliqbl/wallpapers/ui/images/ImagesPagedAdapter.kt]: @adiliqbl Never send a holder object via callbacks as they can be recycled

File: app/src/main/java/com/adiliqbl/wallpapers/util/Util.kt [hunk 5]
```diff
@@ -1,15 +1,5 @@
...
-fun <T> getListFromJson(json: String, clazz: KClass<*>): MutableList<T> {
-    return Moshi.Builder().build()
-            .adapter<MutableList<T>>(Types.newParameterizedType(MutableList::class.java, clazz.java))
-            .fromJson(json)!!
-}
-
-fun toProperCase(s: String): String {
-    return s.substring(0, 1).toUpperCase() + s.substring(1).toLowerCase()
+fun String.properCase(): String {
+    return this.substring(0, 1).toUpperCase() + this.substring(1).toLowerCase()
```
Comment (osamaanwar94) [inline app/src/main/java/com/adiliqbl/wallpapers/util/Util.kt:4]: @adiliqbl The method is redundant as it is just capitalizing the string

Comment (adiliqbl) [reply app/src/main/java/com/adiliqbl/wallpapers/util/Util.kt:4]: I don't think its possible to convert lowercase to sentence case purely with XML. One solution would be to override `setText` method of `TextView`, but I think is much cleaner than creating that.

**Pull request merged**